pkg/cmd/agent-task/create/create_test.go

```diff
@@ -17,13 +20,106 @@ import (
-	// no args should error via cobra MinimumNArgs before our runF
-	// TODO once we support more sources of problem statement input,
-	// this will change.
-	_, err := cmd.ExecuteC()
-	require.Error(t, err)
+	type tc struct {
+		name        string
+		args        []string
+		fileContent string         // if non-empty, create temp file and substitute {{FILE}} token in args
+		wantOpts    *CreateOptions // nil when expecting error
+		expectedErr string
+	}
+
+	tests := []tc{
+		{
+			name:        "no args nor file",
+			args:        []string{},
+			expectedErr: "a task description is required",
+		},
+		{
+			name: "arg only success",
+			args: []string{"task description from args"},
+			wantOpts: &CreateOptions{
+				ProblemStatement: "task description from args",
+			},
+		},
+		{
+			name:        "from-file success",
+			args:        []string{"-F", "{{FILE}}"},
+			fileContent: "task description from file",
+			wantOpts: &CreateOptions{
+				ProblemStatement: "task description from file",
+			},
+		},
+		{
+			name:        "file content from stdin success",
+			args:        []string{"-F", "-"},
+			fileContent: "task from stdin",
+			wantOpts:    &CreateOptions{ProblemStatement: "task from stdin"},
+		},
+		{
+			name:        "mutually exclusive arg and file",
+			args:        []string{"Some task inline", "-F", "{{FILE}}"},
+			fileContent: "Some task",
+			expectedErr: "only one of -F or arg can be provided",
+		},
+		{
+			name:        "missing file path",
+			args:        []string{"-F", "does-not-exist.md"},
+			expectedErr: "could not read task description file: open does-not-exist.md: no such file or directory",
+		},
+		{
+			name:        "empty file",
+			args:        []string{"-F", "{{FILE}}"},
+			fileContent: "   \n\n",
+			expectedErr: "task description file is empty",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ios, stdinBuf, _, _ := iostreams.Test()
+
+			// Provide file content either via stdin ( -F - ) or by creating a temp file
+			if tt.fileContent != "" {
+				isStdin := len(tt.args) == 2 && tt.args[0] == "-F" && tt.args[1] == "-"
+				hasFileToken := slices.Contains(tt.args, "{{FILE}}")
+
+				switch {
+				case isStdin:
+					stdinBuf.WriteString(tt.fileContent)
+				case hasFileToken:
+					dir := t.TempDir()
+					path := filepath.Join(dir, "task.md")
+					if err := os.WriteFile(path, []byte(tt.fileContent), 0o600); err != nil {
+						t.Fatalf("failed to write temp file: %v", err)
+					}
+					for i, a := range tt.args {
+						if a == "{{FILE}}" {
+							tt.args[i] = path
+						}
+					}
+				}
+			}
+
+			f := &cmdutil.Factory{IOStreams: ios}
+			var gotOpts *CreateOptions
+			cmd := NewCmdCreate(f, func(o *CreateOptions) error {
+				gotOpts = o
+				return nil
+			})
+			cmd.SetArgs(tt.args)
+			_, err := cmd.ExecuteC()
+
+			if tt.expectedErr != "" {
+				require.Error(t, err)
+				require.Equal(t, tt.expectedErr, err.Error())
```

Comment:
The error message is different on Windows. That's why the CI is failing.

```
expected: "could not read task description file: open does-not-exist.md: no such file or directory"
actual  : "could not read task description file: open does-not-exist.md: The system cannot find the file specified."
```

I think we can just switch to `require.ErrorContains` like below and remove the OS message part from the test case's `expectedErr`. What do you think?


```suggestion
				require.ErrorContains(t, err, tt.expectedErr)
```